ISPN-14079 Jakarta dependencies depend on non-jakarta ones

jabolina opened the pull request:
https://issues.redhat.com/browse/ISPN-14079

As discussed in Zulip, I've created a new module for the artifacts adding the `-jakarta` postfix. Tristan helped listing the modules we use the eclipse converter and I've created a new module for each, with exception of `extended-statistics`, that even though is using the eclipse plugin it does not have any `javax` use. Should I create a new jakarta module for it too?

Also, comparing against the jar generated in main, the ones in this PR seems to have all the same stuff, but I dont know a good way that we can test it, so suggestions are pretty welcome.

tristantarrant commented:
I've just tried to create a simple project with a single dependency (`org.infinispan:infinispan-client-hotrod-jakarta`) and unfortunately non-jakarta dependencies are still being pulled in:
```
INFO] org.infinispan.playground:hotrod-authentication:jar:0.0.1-SNAPSHOT
[INFO] +- org.infinispan:infinispan-client-hotrod-jakarta:jar:14.0.0-SNAPSHOT:compile
[INFO] |  +- org.infinispan:infinispan-client-hotrod:jar:14.0.0-SNAPSHOT:compile
```
Additionally it is depending on the legacy `jakarta.transaction:jakarta.transaction-api:jar:1.3.3:compile` instead of the new `jakarta.transaction:jakarta.transaction-api:jar:2.0.1:compile`

rhusar commented on cdi/common-jakarta/pom.xml:
```diff
@@ -0,0 +1,68 @@
+    </parent>
+
+    <artifactId>infinispan-cdi-common-jakarta</artifactId>
+    <name>Infinispan CDI JakartaEE support common</name>
```
Kind of a nitpick but "JakartaEE" as one word is not a thing, use "Jakarta EE" instead. Applies to many places. Simple find & replace should do the trick.

jabolina replied:
Updated, thanks!

jabolina commented:
Updated to use a `provided` scope, but I am bit afraid that I am only delegating the need to pull the non-jakarta dependencies to something else. The tree now looks like this:

```text
[INFO] com.jabolina:playground:jar:1.0-SNAPSHOT
[INFO] \- org.infinispan:infinispan-client-hotrod-jakarta:jar:14.0.0-SNAPSHOT:compile
[INFO]    \- org.infinispan:infinispan-commons-jakarta:jar:14.0.0-SNAPSHOT:compile
```

In the last PR I've updated the transaction API, meaning that lots of places needed to update from `javax` to `jakarta`.
Also includes a change removing Geronimo stuff. Looks like there is no support for the transaction version we are using.

rhusar commented on cdi/embedded-jakarta/src/main/java/org/infinispan/cdi/embedded/package-info.java:
```diff
@@ -0,0 +1,4 @@
+/**
+ * Version of Infinispan CDI Embedded using the JakartaEE APIs.
```
@jabolina Let's fix these in Javadoc as well and use "Jakarta EE" instead (there are ~12 occurences).

jabolina commented:
CI is back to normal. Should be good to go now.

tristantarrant commented:
All of the jakarta modules need to have their correct `compile` dependencies added. For example, `hotrod-client-jakarta` should depend on the `netty` components, and so on.

tristantarrant commented:
Additionally, the `cdi-jakarta` module should depend on `jakarta.enterprise.cdi-api` 4.0

tristantarrant commented:
Merged, thanks
